specs/ServerCertificate.md

```diff
@@ -0,0 +1,299 @@
+Server Certificate API
+===
+# Background
+
+The WebView2 team has been asked for an API to intercept when WebView2 cannot verify a server's digital certificate while loading a web page.
+This API provides an option to trust the server's TLS certificate at the application level and render the page without prompting the user about the TLS error or can cancel the request.
+
+# Description
+
+We propose adding `ReceivingServerCertificateError` API that allows you to verify TLS certificates with errors, and either continue the request
+to load the resource or cancel the request.
+
+When the event is raised, WebView2 will pass a `CoreWebView2ReceivingServerCertificateErrorEventArgs` , which lets you view the TLS certificate request Uri, error information, inspect the certificate metadata and several options for responding to the TLS request.
+
+* You can perform your own verification of the certificate and allow the navigation to proceed if you trust it.
+* You can choose to cancel the request.
+* You can choose to display the default TLS interstitial page to let user respond to the request.
```

Comment:
Does this event need a deferral? The app may want to do async work before deciding whether to accept the certificate.

* Prompt the user.
* Send the suspicious certificate to another server for evaluation.
* Perform I/O or database queries to decide whether the certificate is good.

Reply:
It has a deferral in the IDL/MIDL3. Please update the text to note the option to defer.